kit/manifake.man
.\" $Id$
.\"
.\"  Copyright (c) 1991-1997, 2004-2006, Raphael Manfredi
.\"  
.\"  You may redistribute only under the terms of the Artistic Licence,
.\"  as specified in the README file that comes with the distribution.
.\"  You may reuse parts of this distribution only within the terms of
.\"  that same Artistic Licence; a copy of which may be found at the root
.\"  of the source tree for dist 4.0.
.\"
.\" $Log: manifake.man,v $
.\" Revision 3.0.1.1  1995/05/12  11:58:21  ram
.\" patch54: updated my e-mail address
.\"
.\" Revision 3.0  1993/08/18  12:04:33  ram
.\" Baseline for dist 3.0 netwide release.
.\"
.TH MANIFAKE 1 ram
.SH NAME
manifake \- creates a MANIFEST.new out of a MANIFEST file
.SH SYNOPSIS
.B manifake
.SH DESCRIPTION
.I Manifake
is a real simpleminded program that takes a MANIFEST file, such as one
produced by a shar program, and creates a MANIFEST.new file ready to be
used by all the dist programs. What it does is that all archive numbers
are removed and the optional leading comments up to a dashed-line are
thrown away.
.PP
.I Manifake
will typically be used when converting an existing package
to use \fImetaconfig\fR or any other dist utility.
.SH AUTHOR
Raphael Manfredi <Raphael.Manfredi@pobox.com>
.SH FILES
.PD 0
.TP 20
MANIFEST
Original manifest produced by some shar program
.TP
MANIFEST.new
Faked manifest file, suitable for use by any dist programs.
.PD
.SH "SEE ALSO"
makedist(1), metaconfig(1), pat(1).


ChangeLog
Fri Feb 28 17:48:41 MET 1997   Raphael Manfredi <Raphael_Manfredi@grenoble.hp.com>

. Description:

	This huge set of patch results from an integration of the
	perl5 units into the dist-3.0 package. A few improvements
	of mine are also incorporated here, but this patch set should
	mainly be credited to Andy Dougherty and Chip Salzenberg, who
	have both assumed the perl5 Patch Pumpkin with success and brio.

	I won't document most of the changes in detail. Only the obvious
	ones were kept here. You can look closer to the patches themselves
	to see what really changed. Especially the new units, since you
	now have more symbols to check for in your C code...

	Regenerated Configure with current set of units.

	Added Perl Patch Pumpkin credits. Thanks to Andy Dougherty and
	to Chip Salzenberg for their incredible contributions to
	metaconfig.

	Changed my email address and updated copyright dates

	Added support for src.U. This means you can now run Configure
	from a remote directory to separate the build directory and
	the (possibly read-only) source tree.
	Note that some of your private units may require some changes
	due to that...

	Removed support for NO_PROTOTYPE detection on SCO.

	Added cute quoting trick for wild stringify support, which
	allows non-ANSI stringification to occur.

	Replaced .a with $_a all over the place.
	Likewise for .o replaced by $_o.
	You'll have to do that to your own private units.

	Integrated new units from perl5. I will not document most
	of the perl5-related changes here. People interested can
	look at Andy's notes in the perl5 distribution.

	New USE_BSD_GETPGRP to detect the getpgrp() flavour. The old
	USE_BSDPGRP has been obsoleted in favor of USE_BSD_SETPGRP.
	Indeed, setpgrp() and getpgrp() can have distinct flavors,
	so we need two distinct tests for them.

	Useless units dropped: d_group.U and d_passwd.U.

	Mention that <poll.h> can be included if HAS_POLL defined.

	Improved overlapping copy check.

	Added support for Free_t, the type of free().

	Don't use nm with the GNU C library.
	Added support for Linux shared libs.

	Don't prompt them if the void support is high enough for us
	and don't ask for the include path unless they are on a MIPS.

	Added the srcdir support to "configure" now that we have src.U.

	Documented the runnning environment and the src.U unit in
	the metaconfig man page. The $_a and $_o are also mentionned.
	All the generated scripts must now start with a "$startsh"
	to ensure proper shell execution.

	New "create" and "empty" lint directives.

	New -L option for metaxref to match metaconfig and metalint.

	Documents contents of the message sent by patnotify:
	we now let them know the patch priority and description.

	Added a whole bunch of new units, mostly from the perl5 team.

. Files changed:

	* bin/packinit.SH: Fixed one wrong ':' comment in .package.

	* jmake/files/Jmake.rules: Now handles USRINC for dependencies. 
	Smarter about dependencies computation.

	* jmake/files/Jmake.tmpl: Lex path can now be configured.

	* mcon/U/Begin.U: Added Extractall dependency.

	* mcon/U/Config_sh.U, mcon/U/Oldsym.U, mcon/U/Extract.U,
	mcon/U/patchlevel.U, mcon/U/MailAuthor.U, mcon/U/Chk_MANI.U,
	mcon/U/Config_h.U: Added support for src.U.

	* mcon/U/Configdir.U: Have README explicitly mention the package
	name.

	* mcon/U/make.U, mcon/U/d_strtoul.U, mcon/U/Unix.U,
	mcon/U/d_inetaton.U, mcon/U/sitearch.U, mcon/U/d_sanemcmp.U,
	mcon/U/i_locale.U, mcon/U/d_sigaction.U, mcon/U/d_getpgid.U,
	mcon/U/src.U, mcon/U/sitelib.U, mcon/U/d_brokstat.U, mcon/U/man3dir.U,
	, mcon/U/d_sigsetjmp.U, mcon/U/i_sysstat.U, mcon/U/d_strtod.U,
	mcon/U/d_byacc.U, mcon/U/sh.U, mcon/U/Signal.U, mcon/U/d_gnulibc.U,
	mcon/U/man1dir.U, mcon/U/Extractall.U, mcon/U/i_values.U,
	mcon/U/i_sfio.U: Created.

	* mcon/U/Getfile.U: Getfile script now begins with "startsh".

	* mcon/U/Head.U: Make sure we unset CDPATH for shells that support
	this.  Improved Korn shell detection and handling.

	* mcon/U/Instruct.U: Logname / whoami sequence rewritten to use case.

	* mcon/U/Loc.U: Allow users to specify paths on the command line. 
	Will now substitute cp for ln if not supported.

	* mcon/U/Myread.U: Myread script now starts with a "startsh". 
	Miscellaneous fixes.

	* mcon/U/Oldconfig.U: Added support for src.U.  New OSNAME define. 
	Can now sense new OSes.

	* mcon/U/Options.U: Optdef.sh now starts with a "startsh".  Moved
	some code from Head.U.

	* mcon/U/Whoa.U: Whoa script now starts with leading "startsh".

	* mcon/U/afs.U: Can now explicitly tell Configure whether AFS is
	running.

	* mcon/U/archlib.U: Skip existence checks for archlib.

	* mcon/U/archname.U: Changed the way the archname is mangled from
	uname.

	* mcon/U/byteorder.U: No longer ask the user if the test runs ok.

	* mcon/U/ccflags.U: Removed support for NO_PROTOTYPE detection on
	SCO.  New locincpth variable.  Added info on the "additional ld
	flags" question.

	* mcon/U/cf_who.U: New computation method avoiding use of temporary
	file.

	* mcon/U/cpp_stuff.U: Added cute quoting trick for wild stringify
	support.

	* mcon/U/d_bsdjmp.U, mcon/U/d_fd_set.U, mcon/U/models.U,
	mcon/U/d_scannl.U, mcon/U/d_open3.U, mcon/U/d_closedir.U: Added ?F:
	metalint hint.

	* mcon/U/libs.U, mcon/U/libnm.U, mcon/U/libyacc.U, mcon/U/libflex.U,
	mcon/U/libdbm.U, mcon/U/d_socket.U, mcon/U/d_normsig.U,
	mcon/U/d_crypt.U: Replaced .a with $_a all over the place.

	* mcon/U/d_csh.U: Added full_csh to preserve the full path even when
	portable.

	* mcon/U/d_dosuid.U: Moved unit to TOP via a ?Y: layout directive. 
	Tell them /dev/fd is not about floppy disks.

	* mcon/U/d_gconvert.U: Integrated new unit from perl5.

	* mcon/U/d_getpgrp.U: New USE_BSD_GETPGRP to detect the getpgrp()
	flavour.

	* mcon/U/d_group.U, mcon/U/d_passwd.U: Useless unit dropped.

	* mcon/U/d_usendir.U, mcon/U/d_havetlib.U, mcon/U/errnolist.U:
	Replaced .a with $_a all over the place.  Likewise for .o replaced by
	$_o.


kit/manifake.SH
case $CONFIG in
'')
	if test -f config.sh; then TOP=.;
	elif test -f ../config.sh; then TOP=..;
	elif test -f ../../config.sh; then TOP=../..;
	elif test -f ../../../config.sh; then TOP=../../..;
	elif test -f ../../../../config.sh; then TOP=../../../..;
	else
		echo "Can't find config.sh."; exit 1
	fi
	. $TOP/config.sh
	;;
esac
case "$0" in
*/*) cd `expr X$0 : 'X\(.*\)/'` ;;
esac
echo "Extracting kit/manifake (with variable substitutions)"
$spitshell >manifake <<!GROK!THIS!
$startperl
	eval 'exec perl -S \$0 \${1+"\$@"}'
		if \$running_under_some_shell;

# $Id$
#
#  Copyright (c) 1991-1997, 2004-2006, Raphael Manfredi
#  
#  You may redistribute only under the terms of the Artistic Licence,
#  as specified in the README file that comes with the distribution.
#  You may reuse parts of this distribution only within the terms of
#  that same Artistic Licence; a copy of which may be found at the root
#  of the source tree for dist 4.0.
#
# $Log: manifake.SH,v $
# Revision 3.0.1.1  1993/08/19  06:42:18  ram
# patch1: leading config.sh searching was not aborting properly
#
# Revision 3.0  1993/08/18  12:04:32  ram
# Baseline for dist 3.0 netwide release.
#

!GROK!THIS!
$spitshell >>manifake <<'!NO!SUBS!'
$NEWMANI = 'MANIFEST.new';
$MANI = 'MANIFEST';

&manifake;

!NO!SUBS!
chmod 755 manifake
$grep -v '^;#' ../pl/manifake.pl >> manifake
$eunicefix manifake
